client/webserver/site/src/js/wallets.ts

```diff
@@ -1864,19 +1871,54 @@ export default class WalletsPage extends BasePage {
     }
   }
 
+  async getTxHistory (assetID: number, hideMixTxs: boolean, after?: string) : Promise<TxHistoryResult> {
+    let numToFetch = 10
+    if (hideMixTxs) numToFetch = 15
+
+    const res : TxHistoryResult = { txs: [], lastTx: false }
+    let ref = after
+
+    for (let i = 0; i < 40; i++) {
+      const currRes = await app().txHistory(assetID, numToFetch, ref)
+      if (currRes.txs.length > 0) {
```

Comment:
This is kind of a hack. We should be passing a filter all the way in to the wallet.